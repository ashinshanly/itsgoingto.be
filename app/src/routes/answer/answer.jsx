import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { mergeAll } from 'ramda'
import Helmet from 'react-helmet'
import Linkify from 'react-linkify'
import Countdown from 'react-countdown-now'
import { browserHistory } from 'react-router'
import io from 'socket.io-client'
import { fetchPoll, postResponse, APIError } from 'services/api'
import { pollSelector, hasQuestionSelector, totalResponsesSelector, userRespondedSelector,
  updateResponses } from 'store/poll'
import { answersSelector, clearAnswers } from 'store/answers'
import { setLoading, setRequiresPassphrase, requiresPassphraseSelector } from 'store/loader'
import { hasUserSelector } from 'store/user'
import Back from 'components/back'
import Sharing from './components/sharing'
import Answers from './components/answers'
import Passphrase from './components/passphrase'
import './answer.scss'

class Answer extends React.Component {
  componentDidMount = () => {
    const { identifier, hasPoll, setLoading, fetchPoll, clearAnswers, setRequiresPassphrase,
      updateResponses } = this.props

    if (!hasPoll) {
      setLoading(true)
    }
    clearAnswers()
    fetchPoll().then((response) => {
      if (response instanceof APIError) {
        if (response.details.status === 403 && response.details.error.error === 'incorrect-passphrase') {
          setRequiresPassphrase(true)
        } else {
          browserHistory.push('/404')
        }
      }
      setLoading(false)
    })

    this.socket = io(`http://localhost:8001/responses?identifier=${identifier}`)
    this.socket.on('responses-updated', (responses) => {
      updateResponses(responses)
    })
  }

  componentWillUnmount() {
    this.socket.close()
  }

  onResponseSelected = (id) => {
    const { postResponse } = this.props
    postResponse(id).then(() => {
      this.socket.emit('new-response')
    })
  }

  endingToString = (days, hours, minutes, seconds) => {
    let endingString = ''

    endingString += days ? ` ${days} days and` : ''
    endingString += days || hours ? ` ${hours} hours ${!days ? 'and' : ''}` : ''
    endingString += !days && (hours || minutes) ? ` ${minutes} minutes ${!hours ? 'and' : ''}` : ''
    endingString += !days && !hours ? ` ${seconds} seconds` : ''

    return endingString
  }

  countdownRenderer = ({ days, hours, minutes, seconds, completed }) => {
    let ending = ''

    if (completed) {
      ending = 'This poll has now ended'
    } else {
      ending = 'This poll will end in' + this.endingToString(days, hours, minutes, seconds)
    }

    return <span>{ending}</span>
  }

  render () {
    const { hasPoll, poll, requiresPassphrase, answers, totalResponses, userResponded, hasUser } = this.props

    return (
      <div>
        <Helmet>
          <meta charSet='utf-8' />
          <title>{ poll.question }</title>
          <meta name='description' content='Join in the vote and answer this poll at itsgoingto.be' />
          <meta name='keywords' content='question vote poll result' />
        </Helmet>
        { hasPoll &&
          <div>
            <Back />
            <div className='container header-container answer-header-container'>
              <div className='header center-text'>
                <h2><Linkify properties={{ target: '_blank' }}>{ poll.question }</Linkify></h2>
                <Sharing poll={poll} />
                { poll.ended &&
                  <div className='alert alert-success'><span>This poll has now ended</span></div>
                }
                { poll.endDate && !poll.ended &&
                  <div className='alert alert-success'>
                    <Countdown
                      date={poll.endDate.date}
                      renderer={this.countdownRenderer} />
                  </div>
                }
              </div>
            </div>
            <Answers
              poll={poll}
              answers={answers}
              totalResponses={totalResponses}
              userResponded={userResponded}
              onResponseSelected={this.onResponseSelected}
              viewOnly={hasUser} />
          </div>
        }
        { requiresPassphrase &&
          <Passphrase />
        }
      </div>
    )
  }
}

Answer.propTypes = {
  identifier            : PropTypes.string.isRequired,
  poll                  : PropTypes.object.isRequired,
  hasPoll               : PropTypes.bool.isRequired,
  requiresPassphrase    : PropTypes.bool.isRequired,
  answers               : PropTypes.array.isRequired,
  totalResponses        : PropTypes.number,
  userResponded         : PropTypes.bool.isRequired,
  hasUser               : PropTypes.bool.isRequired,
  fetchPoll             : PropTypes.func.isRequired,
  clearAnswers          : PropTypes.func.isRequired,
  setLoading            : PropTypes.func.isRequired,
  setRequiresPassphrase : PropTypes.func.isRequired,
  postResponse          : PropTypes.func.isRequired,
  updateResponses       : PropTypes.func.isRequired
}

Answer.defaultProps = {
  totalResponses : 0
}

const mapStateToProps = (state, props) => ({
  poll               : pollSelector(state, props.params.identifier),
  hasPoll            : hasQuestionSelector(state, props.params.identifier),
  requiresPassphrase : requiresPassphraseSelector(state),
  answers            : answersSelector(state),
  totalResponses     : totalResponsesSelector(state, props.params.identifier),
  userResponded      : userRespondedSelector(state, props.params.identifier),
  hasUser            : hasUserSelector(state),
})

const mapDispatchToProps = (dispatch, props) => ({
  fetchPoll             : () => dispatch(fetchPoll(props.params.identifier)),
  clearAnswers          : () => dispatch(clearAnswers()),
  setLoading            : (value) => dispatch(setLoading(value)),
  setRequiresPassphrase : (value) => dispatch(setRequiresPassphrase(value)),
  postResponse          : (id) => dispatch(postResponse(id, props.params.identifier)),
  updateResponses       : (responses) => dispatch(updateResponses(responses, props.params.identifier))
})

const mergeProps = (stateProps, dispatchProps, ownProps) => mergeAll([stateProps, dispatchProps, ownProps.params])

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)(Answer)
