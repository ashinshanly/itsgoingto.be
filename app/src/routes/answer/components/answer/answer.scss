@import "variables";

.answer-container {
  padding: 0px;

  .options {
    @media(max-width: $screen-sm-max) {
      font-size: 1.25em;
      width: 100%;
    }

    &.show-results {
      .input-options {
        min-width: 68em;
        @media(max-width: $screen-sm-max) {
          width: 100%;
        }
        @media(max-width: $screen-xs-max) {
          min-width: 0em;
        }
      }

      .result-wrapper,
      .input-label-votes {
        opacity: .999;
      }
    }
  }

  .input-options {
    position: relative;
    margin-left: 0px;
    width: auto;
    min-width: 0px;
    transition: min-width 0.4s ease;
  }

  .input-label-options {
    word-break: break-word;
    padding-right: 3.6em;
    padding-top: 0.2em;
    @media(max-width: $screen-sm-max) {
      padding-top: 0.3em;
    }
    padding-bottom: 0.2em;
    line-height: 1.6em;
    @media(max-width: $screen-md-max) {
      line-height: 1.8em;
    }
    @media(max-width: $screen-sm-max) {
      line-height: 2em;
    }
    text-indent: -32px;
    padding-left: calc(1em + 32px);

    &::after {
      position: absolute;
      top: 2px;
      left: 14px;
      width: 40px;
      height: 40px;
      @media(max-width: $screen-md-max) {
        top: -1px;
        left: 9px;
        width: 36px;
        height: 36px;
      }
      @media(max-width: $screen-sm-max) {
        top: 1px;
        left: 8px;
        width: 32px;
        height: 32px;
      }
      border-radius: 50%;
      content: '';
      opacity: 0;
      pointer-events: none;

      border: 2px solid rgba(0,0,0,0.5);

      box-shadow: inset 0 0 0 35px rgba(0,0,0,0);
    }

    &.input-label-options--click::after {
      -webkit-animation: anim-effect-radomir 0.5s ease-out forwards;
      animation: anim-effect-radomir 0.5s ease-out forwards;
    }
  }

  .input-label-votes {
    position: absolute;
    line-height: 4em;
    margin-top: 0px;
    @media(max-width: $screen-md-max) {
      margin-top: 1px;
    }
    @media(max-width: $screen-sm-max) {
      margin-top: 2px;
    }
    right: 2em;
    top: 0;
    color: #9E9E9E;
    opacity: 0;
    transition: opacity 0.3s ease;
  }

  .result-wrapper {
    height: 99%;
    height: calc(100% - 0.6em);

    opacity: 0;
    transition: opacity 0.4s ease;

    margin-top: 3px;
    @media(max-width: $screen-md-max) {
      margin-top: 2px;
    }
    @media(max-width: $screen-sm-max) {
      margin-top: 1px;
    }
    position: absolute;
    top: 0.3em;
    left: 0.6em;
    right: 0.6em;
    z-index: -1;

    box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.06), 0 0 2px rgba(212, 212, 212, 0.5);
    background: $disabled-grey;

    .result {
      position: absolute;
      height: 100%;

      width: 0%;
      transition: width 0.4s ease;

      background: $dark-grey;
      box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.06), 0 0 2px rgba(212, 212, 212, 0.5);
    }
  }
}

@-webkit-keyframes anim-effect-radomir {
  0% {
    opacity: 1;
    -webkit-transform: scale3d(0.4, 0.4, 1);
    transform: scale3d(0.4, 0.4, 1);
  }
  80% {
    box-shadow: inset 0 0 0 2px rgba(0,0,0,0.8);
    opacity: 0.1;
  }
  100% {
    box-shadow: inset 0 0 0 2px rgba(0,0,0,0.8);
    opacity: 0;
    -webkit-transform: scale3d(1.2, 1.2, 1);
    transform: scale3d(1.2, 1.2, 1);
  }
}

@keyframes anim-effect-radomir {
  0% {
    opacity: 1;
    -webkit-transform: scale3d(0.4, 0.4, 1);
    transform: scale3d(0.4, 0.4, 1);
  }
  80% {
    box-shadow: inset 0 0 0 2px rgba(0,0,0,0.8);
    opacity: 0.1;
  }
  100% {
    box-shadow: inset 0 0 0 2px rgba(0,0,0,0.8);
    opacity: 0;
    -webkit-transform: scale3d(1.2, 1.2, 1);
    transform: scale3d(1.2, 1.2, 1);
  }
}
