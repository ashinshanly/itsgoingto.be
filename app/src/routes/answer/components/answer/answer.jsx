import React from 'react'
import PropTypes from 'prop-types'
import { merge } from 'ramda'
import Linkify from 'react-linkify'
import './answer.scss'

export class Answer extends React.Component {
  constructor (props) {
    super(props)
    this.state = { animating : false }
    this.linkClicked = false
  }

  handleClick = () => {
    const { answer, poll, onResponseSelected, viewOnly } = this.props

    if (!this.linkClicked) {
      if (!poll.ended && !viewOnly) {
        this.setState((prevState) =>
          merge(prevState, { animating : true })
        )
        setTimeout(() => {
          this.setState((prevState) =>
            merge(prevState, { animating : false })
          )
        }, 550)

        onResponseSelected(answer.id)
      }
    } else {
      this.linkClicked = false
    }
  }

  linkClick = () => {
    this.linkClicked = true
  }

  calculateWidth = () =>
    (this.props.answer.responsesCount / this.props.totalResponses) * 100 + '%'

  render () {
    const { index, type, checked, answer, poll, viewOnly } = this.props
    const { animating } = this.state
    const width = this.calculateWidth()

    return (
      <span className='input input-options'>
        <span className='result-wrapper'>
          <span className='result' name={'answer-' + index} style={{ width }} />
        </span>
        <input
          id={'answer-' + index}
          name='answer'
          className={type === 'radio'
            ? 'input-radio input-radio-options'
            : 'input-checkbox input-checkbox-options'}
          type={type}
          value={index}
          checked={checked}
          readOnly />
        <label
          htmlFor={'answer-' + index}
          className={'input-label input-label-options' +
                     (animating ? ' input-label-options--click' : '') +
                     (poll.ended || viewOnly ? ' input-label-options--hidden' : '')}
          onClick={this.handleClick}>
          <Linkify properties={{ target: '_blank', onClick: this.linkClick }}>{ answer.answer }</Linkify>
        </label>
        <span htmlFor={'answer-' + index} className='input-label-votes'>
          {answer.responsesCount} votes
        </span>
      </span>
    )
  }
}

Answer.propTypes = {
  index              : PropTypes.number.isRequired,
  type               : PropTypes.string.isRequired,
  answer             : PropTypes.object.isRequired,
  poll               : PropTypes.object.isRequired,
  totalResponses     : PropTypes.number.isRequired,
  checked            : PropTypes.bool.isRequired,
  viewOnly           : PropTypes.bool.isRequired,
  onResponseSelected : PropTypes.func.isRequired
}

export default Answer
