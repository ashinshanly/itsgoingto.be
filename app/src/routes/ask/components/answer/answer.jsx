import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { compose, equals, length, trim } from 'ramda'
import { updateAnswer, removeAnswer } from 'store/answers'
import EventBus from 'components/event-bus'
import './answer.scss'

const KEY_UP_ARROW   = 38
const KEY_DOWN_ARROW = 40
const KEY_BACKSPACE  = 8
const KEY_DELETE     = 46
const KEY_ENTER      = 13

class Answer extends React.Component {
  handleChange = (event) => {
    this.props.onAnswerChange(this.props.index, event.target.value)
  }

  handleKeyPress = (event) => {
    event = event || window.event
    const key = event.keyCode || event.charCode

    switch (key) {
      case KEY_UP_ARROW:
        this.eventBus.emit('focus', this.props.index - 1)
        break
      case KEY_DOWN_ARROW:
      case KEY_ENTER:
        this.eventBus.emit('focus', this.props.index + 1)
        break
      case KEY_BACKSPACE:
      case KEY_DELETE:
        if (compose(equals(0), length, trim)(this.props.text)) {
          event.preventDefault()
          this.props.onRemoveAnswer(this.props.index)
          this.eventBus.emit('focus', this.props.index - 1)
        }
        break
    }
  }

  componentDidMount = () => {
    this.eventBus = EventBus.getEventBus()
    this.eventListener = this.eventBus.addListener('focus', (index) => {
      if (index === this.props.index) {
        this._answer.focus()
      }
    })
  }

  componentWillUnmount = () => {
    this.eventListener.remove()
  }

  render () {
    const { index, text, disabled } = this.props

    return (
      <div className={'input input-answer' + (disabled ? ' input-disabled' : '')}>
        <label
          className='input-label input-label-answer'
          htmlFor={'answer-' + index} >
          {index + 1}
        </label>
        <input
          className='input-field input-field-answer'
          type='text'
          id={'answer-' + index}
          name={'answer-' + index}
          ref={(c) => { this._answer = c; }}
          value={text}
          onChange={this.handleChange}
          onKeyDown={this.handleKeyPress}
          disabled={disabled} />
      </div>
    )
  }
}

Answer.propTypes = {
  index          : PropTypes.number.isRequired,
  text           : PropTypes.string,
  disabled       : PropTypes.bool,
  onAnswerChange : PropTypes.func.isRequired,
  onRemoveAnswer : PropTypes.func.isRequired
}

Answer.defaultProps = {
  text     : '',
  disabled : false,
}

const mapDispatchToProps = (dispatch) => ({
  onAnswerChange : (index, value) => dispatch(updateAnswer(index, value)),
  onRemoveAnswer : (index) => dispatch(removeAnswer(index))
})

export default connect(null, mapDispatchToProps)(Answer)
