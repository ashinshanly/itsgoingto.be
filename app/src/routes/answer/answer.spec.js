/* global expect */
import React from 'react'
import { shallow } from 'enzyme'
import { Answer } from './answer'

const props = {
  identifier: 'Hd2eJ9Jk',
  poll: {
    question: 'Question?',
    ended: false,
    multipleChoice: false,
  },
  hasPoll: true,
  requiresPassphrase: false,
  answers: [],
  totalResponses: 0,
  userResponded: false,
  hasUser: false,
  fetchPoll: jest.fn(() => Promise.resolve()),
  clearAnswers: jest.fn(),
  setLoading: jest.fn(),
  setRequiresPassphrase: jest.fn(),
  postResponse: jest.fn(),
  updateResponses: jest.fn()
}
let wrapper

describe('(Route) answer', () => {
  beforeEach(() => {
    wrapper = shallow(<Answer {...props} />)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('componentDidMount', () => {
    describe('!hasPoll', () => {
      it('should set loading', () => {
        wrapper = shallow(<Answer {...props} hasPoll={false} />)
        expect(props.setLoading).toHaveBeenCalledWith(true)
      })
    })

    it('should clear answers', () => {
      expect(props.clearAnswers).toHaveBeenCalled()
    })

    it('should fetchPoll', () => {
      expect(props.fetchPoll).toHaveBeenCalled()
    })

    describe('poll succesfully fetched', () => {
      it('should set loading false', () => {
        expect(props.setLoading).toHaveBeenCalledWith(false)
      })
    })

    // TODO : Find a way to get these tests to work.
    // describe('fetch poll returns poll not found', () => {
    //   it('should redirect to 404', () => {
    //     browserHistory.push = jest.fn()
    //     const fetchPoll = jest.fn(() => Promise.resolve(new APIError()))
    //     wrapper = shallow(<Answer {...props} fetchPoll={fetchPoll} />)
    //     expect(browserHistory.push).toBeCalledWith('/404')
    //   })
    // })

    // describe('fetch poll returns incorrect passphrase', () => {
    //   it('should call setRequiresPassphrase', () => {
    //     browserHistory.push = jest.fn()
    //     const fetchPoll = jest.fn(() => Promise.resolve(new APIError({
    //       status: 403,
    //       error: { error: 'incorrect-passphrase' },
    //     })))
    //     wrapper = shallow(<Answer {...props} fetchPoll={fetchPoll} />)
    //     expect(props.setRequiresPassphrase).toBeCalledWith(true)
    //   })
    // })
  })

  describe('(Render)', () => {
    describe('!hasPoll and requiresPassphrase', () => {
      it('matches snapshot', () => {
        wrapper = shallow(<Answer {...props} hasPoll={false} requiresPassphrase={true} />)
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('hasPoll', () => {
      it('matches snapshot', () => {
        wrapper = shallow(<Answer {...props} hasPoll={true} />)
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('Poll is ended', () => {
      it('matches snapshot', () => {
        wrapper = shallow(<Answer {...props} poll={{
          question: 'Question?',
          ended: true,
        }} />)
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('Poll has end date', () => {
      it('matches snapshot', () => {
        wrapper = shallow(<Answer {...props} poll={{
          question: 'Question?',
          ended: false,
          endDate: { date: '22/11/2018' },
        }} />)
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})