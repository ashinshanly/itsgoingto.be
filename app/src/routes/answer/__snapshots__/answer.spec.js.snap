// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) answer (Render) !hasPoll and requiresPassphrase matches snapshot 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      Question?
    </title>
    <meta
      content="Join in the vote and answer this poll at itsgoingto.be"
      name="description"
    />
    <meta
      content="question vote poll result"
      name="keywords"
    />
  </HelmetWrapper>
  <withRouter(Connect(Passphrase)) />
</div>
`;

exports[`(Route) answer (Render) Poll has end date matches snapshot 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      Question?
    </title>
    <meta
      content="Join in the vote and answer this poll at itsgoingto.be"
      name="description"
    />
    <meta
      content="question vote poll result"
      name="keywords"
    />
  </HelmetWrapper>
  <div>
    <Back />
    <div
      className="container header-container answer-header-container"
    >
      <div
        className="header center-text"
      >
        <h2>
          <Linkify
            className="Linkify"
            component="a"
            properties={
              Object {
                "target": "_blank",
              }
            }
          >
            Question?
          </Linkify>
        </h2>
        <Sharing
          poll={
            Object {
              "endDate": Object {
                "date": "22/11/2018",
              },
              "ended": false,
              "question": "Question?",
            }
          }
        />
        <div
          className="alert alert-success"
        >
          <Countdown
            controlled={false}
            date="22/11/2018"
            daysInHours={false}
            intervalDelay={1000}
            precision={0}
            renderer={[Function]}
            zeroPadLength={2}
          />
        </div>
      </div>
    </div>
    <withRouter(Connect(Answers))
      answers={Array []}
      onResponseSelected={[Function]}
      poll={
        Object {
          "endDate": Object {
            "date": "22/11/2018",
          },
          "ended": false,
          "question": "Question?",
        }
      }
      totalResponses={0}
      userResponded={false}
      viewOnly={false}
    />
  </div>
</div>
`;

exports[`(Route) answer (Render) Poll is ended matches snapshot 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      Question?
    </title>
    <meta
      content="Join in the vote and answer this poll at itsgoingto.be"
      name="description"
    />
    <meta
      content="question vote poll result"
      name="keywords"
    />
  </HelmetWrapper>
  <div>
    <Back />
    <div
      className="container header-container answer-header-container"
    >
      <div
        className="header center-text"
      >
        <h2>
          <Linkify
            className="Linkify"
            component="a"
            properties={
              Object {
                "target": "_blank",
              }
            }
          >
            Question?
          </Linkify>
        </h2>
        <Sharing
          poll={
            Object {
              "ended": true,
              "question": "Question?",
            }
          }
        />
        <div
          className="alert alert-success"
        >
          <span>
            This poll has now ended
          </span>
        </div>
      </div>
    </div>
    <withRouter(Connect(Answers))
      answers={Array []}
      onResponseSelected={[Function]}
      poll={
        Object {
          "ended": true,
          "question": "Question?",
        }
      }
      totalResponses={0}
      userResponded={false}
      viewOnly={false}
    />
  </div>
</div>
`;

exports[`(Route) answer (Render) hasPoll matches snapshot 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      Question?
    </title>
    <meta
      content="Join in the vote and answer this poll at itsgoingto.be"
      name="description"
    />
    <meta
      content="question vote poll result"
      name="keywords"
    />
  </HelmetWrapper>
  <div>
    <Back />
    <div
      className="container header-container answer-header-container"
    >
      <div
        className="header center-text"
      >
        <h2>
          <Linkify
            className="Linkify"
            component="a"
            properties={
              Object {
                "target": "_blank",
              }
            }
          >
            Question?
          </Linkify>
        </h2>
        <Sharing
          poll={
            Object {
              "ended": false,
              "multipleChoice": false,
              "question": "Question?",
            }
          }
        />
      </div>
    </div>
    <withRouter(Connect(Answers))
      answers={Array []}
      onResponseSelected={[Function]}
      poll={
        Object {
          "ended": false,
          "multipleChoice": false,
          "question": "Question?",
        }
      }
      totalResponses={0}
      userResponded={false}
      viewOnly={false}
    />
  </div>
</div>
`;
